crates/rt/src/thread.rs

```diff
@@ -16,25 +15,30 @@ pub fn wait_for_completion() {
         );
     }
 
-    const MAX_THREADS: usize = 32;
     let cur_thread_id = unsafe { sys::furi_thread_get_current_id() };
     let app_id = unsafe { CStr::from_ptr(sys::furi_thread_get_appid(cur_thread_id)) };
-    let mut thread_ids: [sys::FuriThreadId; MAX_THREADS] = [ptr::null_mut(); MAX_THREADS];
+    let furi_thread_list = unsafe { sys::furi_thread_list_alloc() };
```

Comment:
This looks to me as a leak since the allocation is not accompanied by a call to `furi_thread_list_free`.

Worth mentioning that to make it [panic-safe] it would be right to wrap this into a struct with the specific `Drop` implementation (rather than just call `*_free` at the end of the scope).

[panic-safe]: https://doc.rust-lang.org/nomicon/exception-safety.html

Reply:
Absolutely, I've updated with few wrappers, let's discuss if you find it a better solution.